src/panels/lovelace/common/handle-click.ts

```diff
@@ -34,14 +45,15 @@ export const handleClick = (
       toggleEntity(hass, config.entity!);
       break;
     case "call-service": {
-      if (config.service) {
-        const [domain, service] = config.service.split(".", 2);
-        const serviceData = {
-          entity_id: config.entity,
-          ...config.service_data,
-        };
-        hass.callService(domain, service, serviceData);
+      if (!actionConfig.service) {
+        return;
       }
+      const [domain, service] = actionConfig.service.split(".", 2);
+      const serviceData = {
+        entity_id: config.entity,
```

Comment:
This has caused issues for some services that don't want the `entity_id`. I can't recall them off the top of my head, but should require that the `service_data` be explicitly stated instead?

Reply:
If it doesnt want it, the service should ignore it... I think that should be a service fix

Reply:
Yeah we should stop trying to be smart and adding the entity automatically.

Reply:
We can still automatically populate service data for `toggle`, but not for normal services.